ee_tests/src/specs/quickstart_cheterminal.spec.ts

```diff
@@ -0,0 +1,121 @@
+    support.debug('>>> Go to Edit Profile Page');
+    let editProfilePage = await userProfilePage.gotoEditProfile();
+    support.debug('>>> Go to Edit Profile Page - OK');
+    support.debug('>>> Go to Reset Env Page');
+    let cleanupEnvPage = await editProfilePage.gotoResetEnvironment();
+    support.debug('>>> Go to Reset Env Page - OK');
+
+    await cleanupEnvPage.cleanup(browser.params.login.user);
+
+  });
+
+  afterEach( async () => {
+    await browser.sleep(support.DEFAULT_WAIT);
+
+    support.writeScreenshot('target/screenshots/che_final_' + globalSpaceName + '.png');
+    support.info('\n ============ End of test reached ============ \n');
+    // support.info('\n ============ End of test reached, logging out ============ \n');
+    /* Logout is causing random failures on prod-preview - possibky due to navigating browser windows? */
+    // await dashboardPage.logout();
+  });
+
+  /* Simple test - accept all defaults for new quickstarts */
+
+    // tslint:disable:max-line-length
+  it('Create a new space, new ' + browser.params.quickstart.name + ' quickstart, create Che workspace, run maven in the workspace', async () => {
+    let quickstart = new Quickstart(browser.params.quickstart.name);
+    let spaceName = support.newSpaceName();
+    globalSpaceName = spaceName;
+    let spaceDashboardPage = await dashboardPage.createNewSpace(spaceName);
+
+    let wizard = await spaceDashboardPage.addToSpace();
+
+    support.info('Creating quickstart: ' + quickstart.name);
+    await wizard.newQuickstartProject({ project: quickstart.name });
+    await spaceDashboardPage.ready();
+
+    /* This statement does not reliably wait for the modal dialog to disappear:
+       await browser.wait(until.not(until.visibilityOf(spaceDashboardPage.modalFade)), support.LONGEST_WAIT);
+
+       The above statement fails with this error: Failed: unknown error: Element <a id="spacehome-pipelines-title"
+       href="/username/spaceName/create/pipelines">...</a> is not clickable at point (725, 667). Other element would
+       receive the click: <modal-container class="modal fade" role="dialog" tabindex="-1" style="display:
+       block;">...</modal-container>
+
+       The only reliable way to avoid this is a sleep statement: await browser.sleep(5000); */
+    await browser.sleep(5000);
+
+    /* Open Che display page */
+    await spaceDashboardPage.codebasesSectionTitle.clickWhenReady();
+
+    let spaceChePage = new SpaceChePage();
+    await spaceChePage.createCodebase.clickWhenReady(support.LONGEST_WAIT);
+
+    /* A new browser window is opened when Che opens - switch to that new window now */
+    let handles = await browser.getAllWindowHandles();
+    support.debug('Number of browser tabs before = ' + handles.length);
+
+    handles = await browser.getAllWindowHandles();
+    while (handles.length < 2) {
+        support.info ('***Waiting for Che browser window to open***');
+        await browser.sleep (3000);
```

Comment:
Can't we have waits with some pooling and upper limit instead? In the Java world that would be something like:

```java
WebDriverWait wait = new WebDriverWait(driver, 10);
        wait.pollingEvery(200, TimeUnit.MILISECONDS);
        wait.until(ExpectedConditions.elementToBeClickable(button));
        button.click();
```

Reply:
Thx - I found a way around this.